server/controllers/stock/reports/stock/consumption_graph.js

```diff
@@ -0,0 +1,87 @@
+const {
+  _, db, util, ReportManager, pdfOptions, STOCK_CONSUMPTION_GRAPTH_TEMPLATE,
+} = require('../common');
+
+const FilterParser = require('../../../../lib/filter');
+/**
+   * @method stockEntryReport
+   *
+   * @description
+   * This method builds the stock entry report as either a JSON, PDF, or HTML
+   * file to be sent to the client.
+   *
+   * GET /reports/stock/consumption_graph
+   */
+function stockConsumptionGrathReport(req, res, next) {
+  let report;
+
+  const params = _.clone(req.query);
+
+  const optionReport = _.extend(params, pdfOptions, {
+    filename : 'REPORT.STOCK_CONSUMPTION_GRAPH_REPORT.TITLE',
+    waitFor : 1500,
+  });
+
+  // set up the report with report manager
+  try {
+    report = new ReportManager(STOCK_CONSUMPTION_GRAPTH_TEMPLATE, req.session, optionReport);
+  } catch (e) {
+    return next(e);
+  }
+
+  const sql = `
+    SELECT
+     (m.quantity*m.unit_cost) AS amount, DATE(m.date) as date, i.text
+    FROM stock_movement m
+    JOIN lot l ON l.uuid = m.lot_uuid
+    JOIN inventory i ON i.uuid = l.inventory_uuid
+    JOIN depot d ON d.uuid = m.depot_uuid
+  `;
+  const inventorySql = 'SELECT text FROM inventory WHERE uuid=?';
+  const filterObtions = req.query;
+  db.convert(filterObtions, ['depot_uuid', 'inventory_uuid']);
+  const filters = new FilterParser(filterObtions, { tableAlias : 'm' });
+
+  filters.equals('is_exit');
+  filters.custom('depot_uuid', 'd.uuid=?');
+  filters.custom('inventory_uuid', 'i.uuid=?');
+  filters.dateFrom('dateFrom', 'date');
+  filters.dateTo('dateTo', 'date');
+
+  filters.setGroup(' GROUP BY DATE(m.date) ');
+
+  const rqtSQl = filters.applyQuery(sql);
+  const rqtParams = filters.parameters();
+
+  const bundel = {
+    labels : [],
+    data : [],
+  };
+
+  let inventory = {};
+  return db.one(inventorySql, filterObtions.inventory_uuid).then(invent => {
```

Comment:
Could you use async/await here to clean up this nested structure?

Reply:
No problem.